src/expr.rs

```diff
@@ -282,6 +282,9 @@ pub(crate) fn format_expr(
                 match lhs.kind {
                     ast::ExprKind::Lit(token_lit) => lit_ends_in_dot(&token_lit),
                     ast::ExprKind::Unary(_, ref expr) => needs_space_before_range(context, expr),
+                    ast::ExprKind::Binary(_, ref expr, _) => {
+                        needs_space_before_range(context, expr)
+                    }
```

Comment:
Is this correct? The [ast::ExprKind::Binary](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/enum.ExprKind.html#variant.Binary) is defined as **Binary([BinOp](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/type.BinOp.html), [P](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ptr/struct.P.html)<[Expr](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/struct.Expr.html)>, [P](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ptr/struct.P.html)<[Expr](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/struct.Expr.html)>)**, and I'm assuming that the first expr is the `left expr` (the one that comes before the operator), while the second is the `right expr` (the one that comes after the operator). The way the code is written it looks like you're testing whether the `left expr` needs a space before the range, when I'd expect this to check if the `right expr` needs a space before the range.

Reply:
It's incorrect, you're absolutely right, nice catch! 

Changed to the rhs and added the source example and the other test cases.